### PR TITLE
expose GetMappingRequest to js

### PR DESCRIPTION
**Pull request by sethalves**
- allow javascript to do GetMappingRequests to asset-server

### CODE REVIEW

**Comment by sethalves** [comment]
to test:
- visit a domain where you have write access to the asset-server
- run this javascript:
```javascript
var testData = "this is a test: " + Math.random();
var testFilename = "/asset-test.txt";
Assets.uploadData(testData, function(url, hash) {
    Assets.setMapping(testFilename, hash, function() {
        Assets.getMapping(testFilename, function (retrievedHash) {
            if (retrievedHash != hash) {
                print("FAILURE -- hashes don't match");
            } else {
                Assets.downloadData("atp:" + retrievedHash, function (retrievedData) {
                    if (testData != retrievedData) {
                        print("FAILURE -- data doesn't match");
                    } else {
                        print("SUCCESS");
                    }
                });
            }
        });
    });
});
```
- look for "SUCCESS"

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6288/app_12173_PR.apk

File: libraries/script-engine/src/AssetScriptingInterface.h [hunk 1]
```diff
@@ -75,7 +75,23 @@ class AssetScriptingInterface : public QObject {
...
+     * @function Assets.getMapping
+     * @static
+     * @param path {string}
+     * @param callback {Assets~getMappingCallback}
+     */
+
+    /**jsdoc
+     * Called when getMapping is complete
+     * @callback Assets~getMappingCallback
+     * @param {string} assetID
```
Comment (huffman) [inline libraries/script-engine/src/AssetScriptingInterface.h]: Can we add documentation for the error case?

File: libraries/script-engine/src/AssetScriptingInterface.cpp [hunk 2]
```diff
@@ -53,10 +55,30 @@ void AssetScriptingInterface::setMapping(QString path, QString hash, QScriptValu
...
+        auto result = request->getError();
+        if (callback.isFunction()) {
+            if (result == GetMappingRequest::NotFound) {
+                QScriptValueList args { "" };
+                callback.call(_engine->currentContext()->thisObject(), args);
+            } else if (result == GetMappingRequest::NoError) {
+                QScriptValueList args { request->getHash() };
+                callback.call(_engine->currentContext()->thisObject(), args);
+            } else {
+                qCDebug(scriptengine) << "error -- " << request->getError() << " -- " << request->getErrorString();
```
Comment (huffman) [inline libraries/script-engine/src/AssetScriptingInterface.cpp]: We should make sure to always call the callback, although I know the other functions don't all work this way. What is our preferred method for handling errors in callbacks?

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6288/HighFidelity-Beta-PR12173-3e7d30366125d8ec004ac57c400680c7ca65b2b9.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6288/HighFidelity-Beta-PR12173-3e7d30366125d8ec004ac57c400680c7ca65b2b9.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6293/app_12173_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6293/HighFidelity-Beta-PR12173-76048f3c308d7ca39712c18093c39a16e6eecdc2.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6293/HighFidelity-Beta-PR12173-76048f3c308d7ca39712c18093c39a16e6eecdc2.dmg

**Comment by sbennett77** [comment]
Currently Testing.

**Comment by sbennett77** [comment]
Passed.

@conklin94122

**Pull request merged**